### ISSUE TITLE
DNS actor fails to be created

### ISSUE THREAD

**Issue by drewhk** [akka/akka#18650]
Only affects 2.4. According to user report the below problem happens with invalid hostnames.

Caused by: akka.ConfigurationException: configuration problem while creating [akka://providerintegration/system/IO-DNS/inet-address] with router dispatcher [akka.actor.default-dispatcher] and mailbox [akka.actor.default-mailbox] and routee dispatcher [akka.actor.default-dispatcher] and mailbox [akka.actor.default-mailbox]
at akka.actor.LocalActorRefProvider.actorOf(ActorRefProvider.scala:795) ~[akka-actor_2.11-2.4.0.jar:na]
at akka.remote.RemoteActorRefProvider.actorOf(RemoteActorRefProvider.scala:297) ~[akka-remote_2.11-2.4.0.jar:na]
at akka.actor.dungeon.Children$class.makeChild(Children.scala:220) ~[akka-actor_2.11-2.4.0.jar:na]
at akka.actor.dungeon.Children$class.actorOf(Children.scala:38) ~[akka-actor_2.11-2.4.0.jar:na]
at akka.actor.ActorCell.actorOf(ActorCell.scala:373) ~[akka-actor_2.11-2.4.0.jar:na]
at akka.io.SimpleDnsManager.<init>(SimpleDnsManager.scala:15) ~[akka-actor_2.11-2.4.0.jar:na]
... 18 common frames omitted

**Comment by analytically** [comment]
https://gist.github.com/analytically/63381c0adfee9aeb06b9

**Comment by drewhk** [comment]
So the section is actually in the config. Maybe the actor is restarted because of a failure and some weird thing happens?

**Comment by patriknw** [comment]
doesn't that `akka.ConfigurationException` have a `Caused by` that would tell us what is wrong?

**Comment by analytically** [comment]
Yes:

```
Caused by: akka.ConfigurationException: Configuration missing for router [akka://providerintegration/system/IO-DNS/inet-address] in 'akka.actor.deployment' section.
 	at akka.routing.FromConfig.verifyConfig(RouterConfig.scala:311) ~[akka-actor_2.11-2.4.0.jar:na]
 	at akka.routing.RoutedActorRef.<init>(RoutedActorRef.scala:40) ~[akka-actor_2.11-2.4.0.jar:na]
```

**Comment by kaesler** [comment]
Seeing this one too in my test code. FWIW  Here is a simple example:

import scala.concurrent.Future
import scala.concurrent.duration._

import akka.actor.ActorSystem
import akka.http.scaladsl.Http
import akka.http.scaladsl.client.RequestBuilding
import akka.http.scaladsl.model.Uri
import akka.stream.ActorMaterializer

import org.scalactic.TypeCheckedTripleEquals
import org.scalatest.Matchers
import org.scalatest.concurrent.Futures
import org.scalatest.concurrent.ScalaFutures
import org.scalatest.fixture.FunSpec
import org.scalatest.time.Millis
import org.scalatest.time.Seconds
import org.scalatest.time.Span


class DnsBugSpec
  extends FunSpec
  with Matchers
  with Futures
  with ScalaFutures
  with TypeCheckedTripleEquals {

  // Configure the behavior of the "whenReady" calls below.
  implicit val defaultPatience = PatienceConfig(timeout =  Span(30, Seconds),
                                                interval = Span(1, Millis))
  def await[T](f: Future[T]): T = whenReady(f) { t => t }

  type FixtureParam = ActorSystem

  def withFixture(test: OneArgTest) = {

    val system = ActorSystem()

    try {
      test(system)
    } finally {
      await(system.terminate())
    }
  }

  describe ("The thing") {

    it ("should work") { implicit system =>

     implicit val materializer = ActorMaterializer()

      await(
        Http().singleRequest(
          RequestBuilding.Get(
            uri = Uri("http://roo.bar.foo"))))
    }
  }
}

**Comment by adarshaj** [comment]
Is there any workaround for this?

**Comment by marekzebrowski** [comment]
I can confirm  with 2.4.2 - it happens with dns name that can't be resolved properly:
```
11:29:38 ERROR a.a.OneForOneStrategy: exception during creation
akka.actor.ActorInitializationException: exception during creation
...
Caused by: akka.ConfigurationException: configuration problem while creating [akka://sgActors/system/IO-DNS/inet-address] with router dispatcher [akka.actor.default-dispatcher] and mailbox [akka.actor.default-mailbox] and routee dispatcher [akka.actor.default-dispatcher] and mailbox [akka.actor.default-mailbox]
...
Caused by: akka.ConfigurationException: Configuration missing for router [akka://sgActors/system/IO-DNS/inet-address] in 'akka.actor.deployment' section.
	at akka.routing.FromConfig.verifyConfig(RouterConfig.scala:318) ~[akka-actor_2.11-2.4.2.jar:na]
...
11:29:38 INFO  a.a.RepointableActorRef: Message [akka.io.Dns$Resolve] from Actor[akka://sgActors/system/IO-TCP/selectors/$a/4#-1488701449] to Actor[akka://sgActors/system/IO-DNS#1513889598] was not delivered. [3] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
```

**Comment by patriknw** [comment]
Found it! It's a bug in RemoteActorRefProvider. I'll create PR after lunch.

**Comment by analytically** [comment]
Awesome!

**Comment by patriknw** [comment]
fixed in master and release-2.3
